commit 279b991b24d2439fbe9d2f093988b9c8aed2603d
Author: Matt Redfearn <matt.redfearn@mips.com>
Date:   Thu Mar 31 10:05:36 2016 +0100

    MIPS: Kernel: Add relocate.c
    
    arch/mips/kernel/relocate.c contains the functions necessary to relocate
    the kernel elsewhere in memory
    
    The kernel makes a copy of itself at the new address. It uses the
    relocation table inserted by the relocs tool to fix symbol references
    within the new image.
    
    If copy/relocation is sucessful then the entry point of the new kernel
    is returned, otherwise fall back to starting the kernel in place.
    
    Signed-off-by: Matt Redfearn <matt.redfearn@imgtec.com>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: James Hogan <james.hogan@imgtec.com>
    Cc: Paul Burton <paul.burton@imgtec.com>
    Cc: linux-mips@linux-mips.org
    Cc: kernel-hardening@lists.openwall.com
    Cc: linux-kernel@vger.kernel.org
    Patchwork: https://patchwork.linux-mips.org/patch/12985/
    Signed-off-by: Ralf Baechle <ralf@linux-mips.org>

diff --git a/arch/mips/kernel/Makefile b/arch/mips/kernel/Makefile
index 8a8b775fa7c8..e6053d07072f 100644
--- a/arch/mips/kernel/Makefile
+++ b/arch/mips/kernel/Makefile
@@ -83,6 +83,8 @@ obj-$(CONFIG_I8253)		+= i8253.o
 
 obj-$(CONFIG_GPIO_TXX9)		+= gpio_txx9.o
 
+obj-$(CONFIG_RELOCATABLE)	+= relocate.o
+
 obj-$(CONFIG_KEXEC)		+= machine_kexec.o relocate_kernel.o crash.o
 obj-$(CONFIG_CRASH_DUMP)	+= crash_dump.o
 obj-$(CONFIG_EARLY_PRINTK)	+= early_printk.o